libbeat/cmd/instance/beat.go

```diff
@@ -595,7 +602,7 @@ func (b *Beat) RegisterHostname(useFQDN bool) {
 
 	// state.host
 	stateRegistry := monitoring.GetNamespace("state").GetRegistry()
-	monitoring.NewFunc(stateRegistry, "host", host.ReportInfo(useFQDN), monitoring.Report)
+	monitoring.NewFunc(stateRegistry, "host", host.ReportInfo(hostname), monitoring.Report)
```

Comment:
Depends on https://github.com/elastic/elastic-agent-system-metrics/pull/81 being merged and dependency on `elastic-agent-system-metrics` being bumped up.